tools/toolchains/__init__.py

```diff
@@ -775,9 +778,22 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    @abstractmethod
+    def parse_dependencies(self, dep_path):
```

Comment:
I'm gonig to convert all docstrings to conform to https://www.python.org/dev/peps/pep-0257/